python/tvm/driver/tvmc/autotuner.py

```diff
@@ -0,0 +1,301 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Provides support to auto-tuning networks using AutoTVM.
+"""
+import os.path
+import logging
+import time
+
+from tvm import autotvm
+from tvm import relay
+from tvm.autotvm.tuner import GATuner
+from tvm.autotvm.tuner import GridSearchTuner
+from tvm.autotvm.tuner import RandomTuner
+from tvm.autotvm.tuner import XGBTuner
+
+from . import common, frontends
+from .common import TVMCException
+from .main import register_parser
+
+
+@register_parser
+def add_tune_parser(subparsers):
+    """ Include parser for 'tune' subcommand """
+
+    parser = subparsers.add_parser("tune", help="auto-tune a model")
+    parser.set_defaults(func=drive_tune)
+    parser.add_argument(
+        "--early-stopping",
+        type=int,
+        help="minimum number of trials before early stopping",
+    )
+    parser.add_argument("--hostname", help="hostname or IP address of the host machine")
+    parser.add_argument(
+        "--min-repeat-ms",
+        default=1000,
+        type=int,
+        help="minimum time to run each trial (in milliseconds)",
+    )
+    parser.add_argument(
+        "--model-format",
+        choices=frontends.get_frontend_names(),
+        help="specify input model format",
+    )
+    parser.add_argument(
+        "--number",
+        default=10,
+        type=int,
+        help="number of runs a single repeat is made of. "
+        "The final number of tuning executions is: "
+        "(1 + number * repeat)",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        required=True,
+        help="output file to store the tuning records for the tuning process",
+    )
+    parser.add_argument(
+        "--parallel",
+        default=4,
+        type=int,
+        help="the maximum number of parallel devices to use when tuning",
+    )
+    parser.add_argument("--port", default=9090, type=int, help="the port to connect to")
```

Comment:
Would be better to put it along with the hostname.

Reply:
I agree that merging a host name with optional port in a single parameter, e.g. `hostname[:port]` would be beneficial. However, I think that also adding the key to the same network-related parameter, would be a bit more complicated and obscure, in terms of syntax, making it harder for an end user to grasp.

Based on that, would you agree that for now we keep `hostname[:port]` in a single field, and the keep the key in a separate parameter?

Reply:
Fair enough. Then I'd suggest having a post-processing to check if `rpc-tracker` and `device-key` (need to think about the naming a bit more) are given or none of them. Similar to [this case](https://stackoverflow.com/questions/14350343/python-argparse-either-both-optional-arguments-or-else-neither-one).

Reply:
I aggregated the discussion of remote tuning here and resolve other places to better track the states. Per @FrozenGene 's suggestions of using `--tracker-key` and `--rpc-device-key`, I think `rpc-device-key` is fine, but `tracker-key` is not a key so I would still prefer `rpc-tracker`. I think we can expect them to know the meaning of "tracker" and "device key", because if users need to run RPC runner, they must know how to launch RPC tracker and RPC client in advance.

In summary, here are the action items IMO:
1. Use `--rpc-tracker=hostname[:port]` and `--rpc-device-key=key-name`.
2. Have a checker to make sure both are specified or bot are not specified.
3. Explain how to launch RPC tracker and RPC client in the TVMC tutorial.

@leandron please share your thoughts on this.

Reply:
I think it is good to have a more meaningful name, replacing the generic `--hostname / --port / --tracker-key` with something with more context like `--rpc-****`.

Just in the interest of making the parameters shorter and easier to remember, do you think `--rpc-key` would be acceptable, rather than `--rpc-device-key`? I find that two-word mnemonic are usually simpler to remember. I'm not sure it preserves the whole context you might want to give it.

Example of a command line:
```
tvmc tune --output log1.txt --target="llvm" --rpc-tracker=192.168.0.150:9191 --rpc-key=rasp4b resnet50-v2-7.onnx
```

Reply:
`--rpc-key` is acceptable for me.

Reply:
I pushed a new commit that implements `--rpc-tracker` and `--rpc-key`, as discussed here.